.github/workflows/test.yml

```diff
@@ -100,3 +100,41 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
       - run: 'echo "No build required"'
+
+  e2e:
+    runs-on: ubuntu-latest
+    # dont run e2e tests on forked PRs
+    if: github.repository == 'runatlantis/atlantis'
+    env:
+      TERRAFORM_VERSION: 1.8.3
+      GITHUB_USERNAME: ${{ secrets.ATLANTISBOT_GITHUB_USERNAME }}
+      GITHUB_PASSWORD: ${{ secrets.ATLANTISBOT_GITHUB_PASSWORD }}
+      NGROK_AUTH_TOKEN: ${{ secrets.ATLANTISBOT_NGROK_AUTH_TOKEN }}
+    steps:
+      - uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4
+      - uses: actions/setup-go@cdcb36043654635271a94b9a6d1392de5bb323a7 # v5
+        with:
+          go-version-file: go.mod
+
+      # This version of TF will be downloaded before Atlantis is started.
+      # We do this instead of setting --default-tf-version because setting
+      # that flag starts the download asynchronously so we'd have a race
+      # condition.
+      - uses: hashicorp/setup-terraform@651471c36a6092792c552e8b1bef71e592b462d8 # v3
+        with:
+          terraform_version: 1.8.3
+
+      - name: Setup deps
```

Comment:
Should we consider pinning ngrok as well? Looking at existing CI jobs, we've taken a stance of "pin and immutability where possible" - which this seem to circumvent.

1) Switching from using `ngrok` binary directly to their Docker image; where we can pin down to specific SHA256 hashes (hidden bonus that those images are multi-arch)
2) Use a pinned version of `ngrok` via `https://bin.equinox.io/c/4VmDzA7iaHb/ngrok-3.9.0-linux-amd64.zip` (where `3.9.0` seem to be pattern for all their releases and file names)
3) Install `ngrok` via their APT archive that has pinned support out of the box as well

Reply:
what is the concern of pinning ngrok? to me it is like ubuntu-22.04 in here. If it failed, we just try to fix the e2e flow any way? wdyt

Reply:
I will just give it a go for now.

Reply:
Mostly around "number of variables"; if someone is contributing a new PR and they see E2E fail, and perhaps spending time trying to debug why, and for it to be root-caused to "ngrok broke something" would be quite frustrating contribution experience of 

> changed stuff _here_, but unrelated stuff over _here_ somehow broke